core/net/net.go

```diff
@@ -53,7 +53,7 @@ type API interface {
 
 	// AddReplicator with credentials.
 	// The thread service key and all records will be pushed to paddr.
-	AddReplicator(ctx context.Context, id thread.ID, paddr ma.Multiaddr, opts ...ThreadOption) (peer.ID, error)
+	AddReplicator(ctx context.Context, id thread.ID, paddr ma.Multiaddr, lid peer.ID, opts ...ThreadOption) (peer.ID, error)
```

Comment:
I also considered having this as a TheadOption, but from a client's perspective, this felt messy. It was nicer for 'optional' behavior, but I think the explicit (which log do you want to replicate) behavior is more intuitive